Python/qdk-chemistry/qdk_chemistry/widgets/jsme_widget.py

```diff
@@ -101,38 +135,57 @@ def html_str(self) -> str:
             base_url=BASE_URL,
             js_source=JS_SOURCE,
             var_name=self.name,
-            uid=str(JsmeWidget.n),
+            uid=uid,
             width=self.width,
             height=self.height,
             options=self.options,
             data_type=data_type,
             data=data
         )
 
-    def _repr_html_(self):
-        return self.html_str
+    def _ipython_display_(self):
+        """Display the widget"""
+        uid = self._gen_uid()
+        jsme = HTML(self.html_str(uid=uid))
+        button = JsmeWidgetButton(uid=uid)
+        
+        def _save(button):
+            self.save(button.uid)
+        
+        button.on_click(_save)
+        display(jsme, button)
+
+    def save(self, uid):
+        """Save the value of the molecular diagram drawn in the widget to the Python namespace
+        """
+        display(Javascript(f"saveAs_{uid}('w')"))
 
     def to_mol(self, add_hs: bool=False, num_confs: int=10) -> "Mol":
         """Convert widget value to RDKit molecule.
         If Hydrogen atoms are added, calculate the optimal conformer to get H coordinates.
 
-        Args:
-            add_hs (bool, optional): Add Hydrogen atoms
-            num_confs (int, optional): Number of conformers to generate
-
-        Returns:
-            Mol: RDKit Mol object
+        :param add_hs: Add Hydrogen atoms
+        :type add_hs: bool
+        :param num_confs: Number of conformers to generate
+        :type num_confs: int
+        :return: RDKit molecule object
+        :rtype: Mol
         """
-        if self.value.smiles is not "":
+        if self.value.smiles != "":
```

Comment:
Since the empty string is the only false-y string, you may be able to replace:

```suggestion
        if self.value.smiles:
```

Reply:
I strongly favor explicit over implicit; for example, if someone would set `smiles` to `False` or `0`, the above would not pass. Being explicit consistently can avoid these kinds of edge cases.